File: problems/22/solution.py
class Solution:
    def findMin(self, A):
        lo = 0
        hi = len(A) - 1

        # 5, 1, 2, 3, 4
        while A[lo] > A[hi]:
            mid = (lo + hi) / 2
            if A[mid] > A[hi]:
                lo = mid + 1
            else:
                hi = mid
        return A[lo]

if __name__ == "__main__":
    test_cases = int(raw_input())
    for i in range(test_cases):
        lineItems = [int(l) for l in raw_input().split(" ")]
        N = lineItems[0]
        inputArray = lineItems[1:]

        s = Solution()
        result = s.findMin(inputArray)
        print result

File: problems/22/solution.c
#include <stdio.h>
#include <stdlib.h>

int findMin(const int* A, int N) {
    
}

int main() {
	setbuf(stdout, NULL);
	int testcases;
	scanf("%d", &testcases);
	int t = 0;
	for (t = 0; t < testcases; t++) {
		int n; scanf("%d", &n);
        int input[n];
        for (int i = 0; i < n; i++) {
            scanf("%d", &input[i]);
        }
        int K; scanf("%d", &K);
        int result = search(input, n, K);
		printf("%d\n", result);
	}
	return 0;
}
